pkg/logs/launchers/integration/launcher.go

```diff
@@ -0,0 +1,202 @@
+			filepath, err := s.createFile(source)
+			if err != nil {
+				ddLog.Warn("Failed to create integration log file: ", err)
+				continue
+			}
+
+			filetypeSource := s.makeFileSource(source, filepath)
+			s.sources.AddSource(filetypeSource)
+
+			// file to write the incoming logs to
+			s.integrationToFile[source.Name] = filepath
+		case log := <-s.integrationsLogsChan:
+			integrationSplit := strings.Split(log.IntegrationID, ":")
+			integrationName := integrationSplit[0]
+			filepath := s.integrationToFile[integrationName]
+
+			err := s.ensureFileSize(filepath)
+			if err != nil {
+				ddLog.Warn("Failed to get file size: ", err)
+				continue
+			}
+
+			err = s.writeFunction(filepath, log.Log)
+			if err != nil {
+				ddLog.Warn("Error writing log to file: ", err)
+			}
+		case <-s.stop:
+			return
+		}
+	}
+}
+
+// writeLogToFile is used as a function pointer
+func writeLogToFile(filepath, log string) error {
+	file, err := os.OpenFile(filepath, os.O_WRONLY, 0644)
+	if err != nil {
+		ddLog.Warn("Failed to open file to write log to: ", err)
+		return err
+	}
+
+	defer file.Close()
+
+	_, err = file.WriteString(log)
+	if err != nil {
+		ddLog.Warn("Failed to write integration log to file: ", err)
+		return err
+	}
+
+	return nil
+}
+
+// makeFileSource Turns an integrations source into a logsSource
+func (s *Launcher) makeFileSource(source *sources.LogSource, filepath string) *sources.LogSource {
+	fileSource := sources.NewLogSource(source.Name, &config.LogsConfig{
+		Type:        config.FileType,
+		TailingMode: source.Config.TailingMode,
+		Path:        filepath,
+		Name:        "integrations",
+		Source:      "integrations source",
+		Tags:        source.Config.Tags,
+	})
+
+	return fileSource
+}
+
+// TODO Change file naming to reflect ID once logs from go interfaces gets merged.
+// createFile creates a file for the logsource
+func (s *Launcher) createFile(source *sources.LogSource) (string, error) {
+	directory, filepath := s.integrationLogFilePath(*source)
+
+	err := os.MkdirAll(directory, 0755)
+	if err != nil {
+		return "", err
+	}
+
+	file, err := os.Create(filepath)
+	if err != nil {
+		return "", nil
+	}
+	defer file.Close()
+
+	return filepath, nil
+}
+
+// integrationLogFilePath returns a directory and file to use for an integration log file
+func (s *Launcher) integrationLogFilePath(source sources.LogSource) (string, string) {
+	fileName := source.Name + ".log"
+	directoryComponents := []string{s.runPath, "integrations", source.Config.Service}
+	directory := strings.Join(directoryComponents, "/")
+	filepath := strings.Join([]string{directory, fileName}, "")
+
+	return directory, filepath
+}
+
+// ensureFileSize enforces the max file size for files integrations logs
+// files. Files over the set size will be deleted and remade.
+func (s *Launcher) ensureFileSize(filepath string) error {
+	maxFileSizeSetting := pkgConfig.Datadog().GetInt("logs_config.integrations_logs_files_max_size")
```

Comment:
not that I think we will ever overflow it, can use `GetInt64` here to avoid the cast below.

Reply:
Changed